haystack/components/generators/chat/hugging_face_local.py

```diff
@@ -33,6 +36,41 @@
 
 PIPELINE_SUPPORTED_TASKS = ["text-generation", "text2text-generation"]
 
+DEFAULT_TOOL_PATTERN = (
+    r"(?:<tool_call>)?"
+    r'(?:\s*\{.*?"name"\s*:\s*"([^"]+)".*?"arguments"\s*:\s*(\{[^}]+\}).*?\}'
+    r'|\{.*?"function"\s*:\s*\{.*?"name"\s*:\s*"([^"]+)".*?"arguments"\s*:\s*(\{[^}]+\}).*?\})'
+)
+
+
+def default_tool_parser(text: str) -> Optional[List[ToolCall]]:
+    """
+    Default implementation for parsing tool calls from model output text.
+
+    Uses DEFAULT_TOOL_PATTERN to extract tool calls.
+
+    :param text: The text to parse for tool calls.
+    :returns: A list containing a single ToolCall if a valid tool call is found, None otherwise.
```

Comment:
we are only extracting a single tool call, but this is clearly explained.
We can improve this in future if there are requests from the community.

Reply:
Yes exactly, that's why I left it as a list 💪